docs/testing/e2e-plan.md

```diff
@@ -0,0 +1,79 @@
+
+This manifest will be generated in response to a pull request change or commit to selected branches (main, release-x, etc) of the eks-anywhere-packages, eks-anywhere, and eks-anywhere-build-tooling repositories (the last of which possibly limited to changes affecting a curated packages project), and will in turn be used by the testing framework.
+
+#### Increased Frequency Of E2E Test Runs
+
+If feasible, end-to-end tests will be run on each (ok-to-test) pull request on the eks-anywhere-packages repository. If that isn't feasible, then hourly or semi-hourly test runs will help catch problems sooner.
+
+Aside: The quality of curated packages relies upon the quality of our end-to-end tests. Correspondingly, the more often we run them, the faster we learn of, and can fix, problems. If we're not able to run the full battery of tests for each PR, then it is strongly recommended that we do whatever is in our power to be able to do so.
+
+#### Test Configuration Control
+
+A means by which tests can be disabled or enabled based upon their properties, requirements, or current statuses. In plainer words, a way to write a test that doesn't run for a specific provider, or to temporarily disable tests for specific host OS. This input, along with the test manifest are envisioned as being the bulk of the input to a given CP E2E test run.
+
+#### End-To-End Test Documentation
+
+Documentation will be generated that, in addition to the forthcoming design document, will aid developers and other contributors understand the end-to-end test framework, specifically how to write a new end-to-end test, how to run an end-to-end test outside of the build pipelines, how to be notified of failures, and how to debug test failures.
+
+A stretch goal could be to add documentation-style tests in the spirit of [Rust's documentation tests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) or [go test examples](https://pkg.go.dev/testing#hdr-Examples).
+
+#### Independent Build Tooling and Testing
+
+In order for new CP end-to-end tests to be as straightforward as possible, and to run with the desired frequency goals, it is necessary to separate our build-tooling processes from those currently existing in the eks-anywhere and eks-anywhere-build-tooling repositories. The current processes produce significant developer friction when writing tests as well as running tests outside of the build pipelines. The existing system's builds are not as finely-grained as is required to meet the test-run-frequency goals defined previously in this document. Lastly, because the concerns of CP tests have little overlap with those of the EKS-A CLI, it is deemed appropriate that these distinctions in build and testing frameworks would allow each project to move more freely in diretions that serves their individual goals best, while any infrastruture or utilities common to both could live in a neutral repository and shared by each.
```

Comment:
Do we need to separate from build-tooling? We could have a seperate process that for example builds main daily so our E2E test runs agains that. We could then continue to use build tooling to cut git tags which specifies that a build is production ready.

Reply:
Long-term, we plan to build far more frequently than just daily, and our builds really shouldn't be tied to or limited by the existing built-tooling tagging system. The tag system is an artifact of how EKS-A is built from a separate repo from the dev repo. I'm sure they have good reasons for that, but by developing and building from the same repo, we simplify our build system. It's easier to trace what's being built, and to ensure that it's as reproducible as possible.